### DISCUSSION TITLE
[Simulation question]: Problem in Definition of Volumetric Source

### DISCUSSION THREAD

**Discussion by SFR1993** [topas-nbio/TOPAS-nBio#42] [Simulation-question] [answered]
### Simulation question

Hello.
I tried to define a volumetric ring source, but I faced with a problem. I wanted to concentrate(focus) the radiation in z-axis, but it distributed from 0 to 90 degree! Also, although I changed the cut-off and angular distribution, there were no differences in radiation distributions! It seems that in my modification, these lines do not affect the simulation! Could you please help me to solve this problem?

Here is a part of my code modifications: (I am using OpenTOPAS.)
...
# Ring Source Geometry
s:Ge/RingSource/Type 		= "TsCylinder"  # Define as a cylinder
s:Ge/RingSource/Parent 		= "World"
dc:Ge/RingSource/TransZ 	= -6.0 um       # Positioning along Z
dc:Ge/RingSource/RMax 		= 1.0 um        # Outer radius of the ring
dc:Ge/RingSource/Rmin 		= 0.5 um    # Inner radius of the ring
dc:Ge/RingSource/HL 		= 0.01 nm              # Half-length along Z
sc:Ge/RingSource/Material 	= "G4_Pb"   # Set the material
s:Ge/RingSource/Color 		= "white"
...
# Source Definition
s:So/MySource/Type = "Volumetric" 
s:So/MySource/Component = "RingSource" 
s:So/MySource/ActiveMaterial = "G4_Pb" 
s:So/MySource/BeamParticle = "e-" 
d:So/MySource/BeamEnergy = 1 MeV
u:So/MySource/BeamEnergySpread = 0.0
s:So/MySource/BeamPositionDistribution = "Gaussian"    # also I tried it with "Uniform"

s:So/MySource/BeamPositionCutoffShape = "Ellipse" 
d:So/MySource/BeamPositionCutoffX = 1.0 um 
d:So/MySource/BeamPositionCutoffY = 1.0 um 
d:So/MySource/BeamPositionSpreadX = 0.01 um
d:So/MySource/BeamPositionSpreadY = 0.01 um

s:So/MySource/BeamAngularDistribution = "Gaussian"    # also I tried it with "Uniform"
d:So/MySource/BeamAngularCutoffX = 45.0 deg    # also I tried it for 0 to 90 degree
d:So/MySource/BeamAngularCutoffY = 45.0 deg    # also I tried it for 0 to 90 degree
d:So/MySource/BeamAngularSpreadX = 0.1 rad
d:So/MySource/BeamAngularSpreadY = 0.1 rad
i:So/MySource/NumberOfSourcePoints = 100
i:So/MySource/NumberOfHistoriesInRun = 10

### TOPAS-nBio category

Geometry

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by JoseRamosMendez** [comment, marked as answer]
Instead of `volumetric`, use TOPAS-nBio's `VolumeOrSurface` [see here](https://github.com/topas-nbio/TOPAS-nBio/tree/master/primary/TsGeneratorUniformlyRandomAtVolume.cc). 

    s:So/mySource/Type = "VolumeOrSurface"
    s:So/mySource/Component = "myComponent"
    s:So/mySource/DistributePrimariesAtVolumeRegion = "surface"
    s:So/mySource/DirectPrimariesTo = "inside"

But for focusing strictly on the z-axis, then you have to change the code a bit. For that, in line 112 of `TsGeneratorUniformlyRandomAtVolume.cc` set (e.g., all at x=0, y=0 uniformly along z-axis):

    insideVolume = G4ThreeVector(0,0,z);

Best.

**Comment by SFR1993** [comment]
Dear Dr. Ramos Mendez,
Thank you for your reply. I corrected _line 112_. Then I ran the program once with _Volumetric_ mode (1st picture) and once with _VolumeOrSurface_ mode (2nd photo). In both cases, changing the line 112 does not affect the radiation direction; whether only x is not zero, y or z! (@JoseRamosMendez )
![Volumetric_modifiedLine112](https://github.com/user-attachments/assets/5c05620a-b76b-4981-9978-8397fbac8094)
![VolumeOrSurface_modifiedLine112](https://github.com/user-attachments/assets/e6035545-4183-4e9d-9c96-51095c46fef6)

**Reply by JoseRamosMendez** [reply]
Did you rebuild Topas? Each time you change .cc/.hh code, you must rebuild using `make; make install`

**Reply by SFR1993** [reply]
I had not paid attention to this point. It works correctly now.
I appreciate your consideration and guidance.
